doc/design/networking/Investigation-on-NSX-integration.md

```diff
@@ -0,0 +1,254 @@
+     * POST /2.0/services/securitygroup/bulk/{scopeId} (For the scopeId use globalroot-0 for non-universal security groups and universalroot-0 for universal security groups.)
+  * update security policies to apply them to the security group
+     * PUT /2.0/services/policy/securitypolicy/{ID}
+
+* docker network delete 
+  * check if any VMs attached to this security group
+     * GET /2.0/services/securitygroup/{objectId}/translation/virtualmachine 
+  * delete the network (security group, policy group)
+     * DELETE /2.0/services/securitytags/tag/{tagId}
+     * DELETE /2.0/services/securitygroup/{objectId} 
+
+* docker network list
+  * list all the security groups
+    * GET /2.0/services/securitygroup/internal/scope/{scopeId}
+  * filter security group names by `<vch name>-SG-` prefix
+
+* docker run --net=<network name> <image name>
+  * update security group `<vch name>-SG-<network name>` dynamic membership 
+    * PUT /2.0/services/securitygroup/bulk/{objectId}
+
+
+## Discussion:
+
+### Bridged Containers with Exposed Port or Directly Connected to Public network
+  * all the user cases mentioned in the networking design document should still be valid 
+
+## Unclear
+### Adding a ContainerVM to Multiple Networks
+This is not supported in current VIC implementation. 
+  * Is adding another NIC to a containerVM the right way to do it?
+  * Does having multiple interfaces matter ? How would user use multiple NICs?
+
+### Details of Integrating with Current Port Layer API 
+  * does the current Port Layer API map perfectly to our goals in using NSX ?
+  * if not, what changes do we need to make?
+
+### With or Without NSX?
+Do we surpport both cases :
+  1. User does not have NSX setup, then we continue providing with current VIC
+  2. User has NSX setup, then we provide VIC with NSX integration
+  * need to find out how to verify if a bridge network specified is a logic switch?
+
+### User Management
+  * NSX allows RBAC, we can add Vcenter users and assign roles for them
+  * Only security Admin or enterprise admin can operate on NSX security groups and policies. 
+  * Is there a way to assign users to a logical switch?
+
+### Approaches Comparison: Security Tag versus vNIC versus Logical Switch
+  * Security Group Membership with Security Tag: 
+    * Pros:
+      * We only need to tag a VM then it is added to a security group. 
+      * we do not need to update security groups everytime a new containerVM is created. 
+    * Cons:
+      * It seems at the backend of NSX, IP address is used to identify the membership of a security group. So it does not work in this scenario, for example, when there are two port groups on the same VDS and each of them have a VIC (say VIC-a and VIC-b) created. Then the security group and policy on VIC-a aslo affects VIC-b. (One NSX-v tenant is not allowed to have overlapping IPs. But VIC users might be the same tenant (if our understanding of a NSX tenant is correct).)
```

Comment:
RE:  IP mapping to SG.  I'm really surprised by this.  You can't tag by something more unique like mac or MoRef?